### PR TITLE
allow niftyreg to be built with opencl

### PR DESCRIPTION
**Pull request by rijobro**

### CODE REVIEW

**Comment by KrisThielemans** [comment]
Ideally some of this should be in their export file (also for openmp). I guess it isn't?

**Comment by rijobro** [comment]
No, just flags saying that NiftyReg was built with those libraries

File: src/Registration/cReg/CMakeLists.txt [hunk 1]
```diff
@@ -58,6 +58,15 @@ if (NIFTYREG_BUILT_WITH_OPENMP)
   endif()
 endif()
 
+# If niftyreg was bulit with OpenCL
+if (NIFTYREG_BUILT_WITH_OPENCL)
+  find_package(OpenCL REQUIRED)
+  if (OpenCL_FOUND)
```
Comment (KrisThielemans) [inline src/Registration/cReg/CMakeLists.txt]: can dispense with that, as it's `REQUIRED`

File: src/Registration/cReg/CMakeLists.txt [hunk 2]
```diff
@@ -58,6 +58,15 @@ if (NIFTYREG_BUILT_WITH_OPENMP)
...
+    TARGET_LINK_LIBRARIES(Reg PUBLIC ${OpenCL_LIBRARIES})
+    INCLUDE_DIRECTORIES(${OpenCL_INCLUDE_DIRS})
```
Comment (KrisThielemans) [inline src/Registration/cReg/CMakeLists.txt]: ideally also target-specific, but ok

**Comment by KrisThielemans** [comment]
@rijobro merge in master to see

**Comment by rijobro** [comment]
merging as it looks fine to me (plus I doubt Travis has openCL anyway).

**Pull request merged**